Framework/Core/src/runDataProcessing.cxx

```diff
@@ -1138,7 +1138,7 @@ int runStateMachine(DataProcessorSpecs const& workflow,
   uv_tcp_t serverHandle;
   serverHandle.data = &serverContext;
   uv_tcp_init(loop, &serverHandle);
-  driverInfo.port = 8080;
+  driverInfo.port = 8080 + (getpid() % 30000);
```

Comment:
I have tried this change locally before, and it successfully mitigates the issue, although it is obviously still possible but extremely unlike to get collisions.
Perhaps the abort criterion in the port finder should be changed slightly, such that each process wil scan the full port range.

Reply:
agreed. But I'd rather fix it for real, though. I do not see any reason why this should break in the first place. Maybe linux is slow in giving back the ports after a process has exited?

Reply:
I agree that we should fix this for good, but this mitigation in any case makes sense. It also slightly speeds up the port-finding :)
I don't believe it has anything to do with delays. I have played around a bit and put delays of seconds, and it didn't change the behavior. To me it seems that once it got to this state where it collided on the port, it goes to some weird state from which it will never recover.
What I absolutely fail to understand: Why does my small testcase in https://alice.its.cern.ch/jira/browse/O2-2014 not reproduce the issue? It runs the exact same port finding. So the issue must also be related to something else.

Reply:
yes, it's definitely weird and I think there is something fundamental I miss on how libuv does it. I am also confused on why you cannot reproduce the issue with a simple test...